### ISSUE TITLE
Should .files pretend virtual overview files don't exist?

### ISSUE THREAD

**Issue by takluyver** [European-XFEL/EXtra-data#286]
The `.files` attribute (on DataCollection, SourceData, KeyData) is not documented, but also not `_private` by convention. This puts it in a grey area as API.

It still technically works with virtual overview files - it will point to one virtual overview file (per run) rather than a list of actual source files. But for some purposes, people might want to see a list of the relevant source files.

It's relatively easy to get a list of all the source files for a run - either from the virtual overview file itself, or by scanning the run folder. We could make `.files` a property that does this on demand. But it gets trickier to filter it following source & train selections if EXtra-data is not actually using the files. Still possible if we have to, but trickier.

A workaround is that people can pass `_use_voview=False` when opening a run to bypass the virtual overview file and work with the regular source files like before.

**Comment by philsmt** [comment]
Does `_use_voview=False` continue to use the legacy data map mechanism? Are they also written when opening with `_use_voview=True` the first time?

**Comment by takluyver** [comment]
With `_use_voview=False`, the JSON 'run data map' files are created and used as before. If it can use a virtual overview file, the JSON map is neither used nor created.

https://github.com/European-XFEL/EXtra-data/blob/5c2046f6385b3a3c767f2e85ab97765d8dbc2d4d/extra_data/reader.py#L1419-L1432

**Comment by philsmt** [comment]
One cautious option might be to rename the current property to `_files`, and throw an exception for `files` if opened with virtual overview files. That at least protects against code using it from failing silently, and we might decide to simulate the previous behaviour as you described in a second step.

**Comment by takluyver** [comment]
True, but depending on what code wants `.files` for, it may also break things that currently work. The list of one `FileAccess` object is valid.

**Comment by philsmt** [comment]
That is true, but my worry is silently breaking code exactly because the list is still valid, but not what was expected. Preventing that may well be worth it visibly breaking code that may have worked anyways, but which can be fixed as easily.

**Comment by takluyver** [comment]
But the easiest fix for code where that's not an issue would be to use `._files` instead, which trains people to use private attributes and makes future changes more difficult. Or they disable using virtual overview files, and then don't get the benefits.

**Comment by tmichela** [comment]
I actually bumped into it today and I used `_use_voview=False` as a workaround.

we could maybe populate the list with original files when using overview files (the path information to original files is stored there) and lazily generate `FileAccess` whenever trying to access `DataCollection.files`. But honestly I'm not sure it's worth the effort...

Are you aware of any practical use of `.files`?
